ipatests/test_integration/test_trust.py

```diff
@@ -463,6 +357,50 @@ def test_user_uid_resolution_in_external_trust_with_shared_secret(self):
         assert re.search(
             testuser_regex, result.stdout_text), result.stdout_text
 
+    def test_override_homedir(self):
+        """
+        https://pagure.io/SSSD/sssd/issue/2474
+        POSIX attributes, such as shell or home directory, should not be overwritten or missing.
+        """
+        tasks.establish_trust_with_ad(self.master, self.ad_domain,
+                                      extra_args=['--range-type', 'ipa-ad-trust'])
+
+        tasks.backup_file(self.master, paths.SSSD_CONF)
+        testuser = 'testuser@%s' % self.ad_domain
+        result1 = self.master.run_command(['getent', 'passwd', testuser])
+        assert "/home/%s/testuser" % self.ad_domain in result1.stdout_text
+
+        domain = self.master.domain
+        tasks.modify_sssd_conf(
+            self.master,
+            domain.name,
+            {
+                'subdomain_homedir': '%o'
+            }
+        )
+        sssd_conf = self.master.get_file_contents(
+            paths.SSSD_CONF, encoding='utf-8'
+        )
+        assert "subdomain_homedir" in sssd_conf
+        tasks.clear_sssd_cache(self.master)
+
+        run_cmd = '"import os; os.initgroups({0}, 1234)"' .format(testuser)
```

Comment:
This looks ugly to me. Could you please ask Jakub Hrozek if there is some other way to trigger initgroups? Maybe something like "sudo -u testuser" will work.